types/src/primary.rs

```diff
@@ -520,21 +520,37 @@ pub struct BatchMessage {
     pub transactions: Batch,
 }
 
+// TODO: refactor BlockRemoverResult and BlockResult into one type shared by get/remove collections.
```

Comment:
The `block_remover` and the `block_waiter` were not really designed to be aligned on their responses - maybe there is an opportunity to do so - since they are having a different role. I have no problem trying to get those aligned, but let's keep in mind that we don't want to create too much dependency.

Reply:
That is fair, I will remove this todo. Do you think we should still return one error per collection in the `block_remover` case? Currently with everything being handled in batches is it possible for some collections to be deleted and some to fail?